### PR TITLE
Make Ed25519HdWallet/Secp256k1HdWallet usable in profile

### PR DESCRIPTION
**Pull request by webmaster128**

### CODE REVIEW

File: packages/iov-keycontrol/src/userprofile.ts [hunk 1]
```diff
@@ -191,9 +191,9 @@ export class UserProfile {
   }
 
   // creates an identitiy in the n-th keyring entry of the primary keyring
-  public async createIdentity(id: number | KeyringEntryId): Promise<LocalIdentity> {
+  public async createIdentity(id: number | KeyringEntryId, options?: any): Promise<LocalIdentity> {
```
Comment (ethanfrey) [inline packages/iov-keycontrol/src/userprofile.ts]: not so happy with the `any|undefined` type, as mentioned in another issue (let me know if you want me to find the link), but let's clean that up in 0.7. For now we need for compatibility.

File: packages/iov-keycontrol/src/userprofile.spec.ts [hunk 2]
```diff
@@ -4,12 +4,16 @@ import MemDownConstructor from "memdown";
 import { ReadonlyDate } from "readonly-date";
 
 import { Address, Nonce, PrehashType, SendTx, SignableBytes, SignedTransaction, SigningJob, TokenTicker, TransactionIdBytes, TransactionKind, TxCodec } from "@iov/bcp-types";
+import { Slip10RawIndex } from "@iov/crypto";
+import { Encoding } from "@iov/encoding";
 import { Algorithm, ChainId, PostableBytes, PublicKeyBytes, SignatureBytes } from "@iov/tendermint-types";
 
 import { Keyring, KeyringEntryId } from "./keyring";
-import { Ed25519SimpleAddressKeyringEntry } from "./keyring-entries";
+import { Ed25519SimpleAddressKeyringEntry, Secp256k1HdWallet } from "./keyring-entries";
```
[the remark this answers is not in the document]
Comment (webmaster128) [reply packages/iov-keycontrol/src/userprofile.spec.ts]: The test added in this file does not aim to test all entry types but to test that the identity options are passed around properly. To test this, I only need one example that uses options.

File: packages/iov-keycontrol/src/keyring.ts [hunk 3]
```diff
@@ -56,10 +61,12 @@ export class Keyring {
 
   private static readonly deserializationRegistry = new Map([
     ["ed25519", (data: KeyringEntrySerializationString) => new Ed25519KeyringEntry(data)],
+    ["ed25519-hd", (data: KeyringEntrySerializationString) => new Ed25519HdWallet(data)],
```
[the remark this answers is not in the document]
Comment (webmaster128) [reply packages/iov-keycontrol/src/keyring.ts]: yeah, there is registerEntryType just obove that variable

File: packages/iov-keycontrol/src/userprofile.ts (hunk 1 above)
Comment (webmaster128) [reply packages/iov-keycontrol/src/userprofile.ts]: I am also not exactly happy. But I prefer to use any until we know what kind of options we'll have in 0.7. First step is to make options non-optional and then we see what is left

File: packages/iov-keycontrol/src/keyring.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ethanfrey) [reply packages/iov-keycontrol/src/keyring.ts]: "read the source luke"....

**Pull request merged**